lib/printers/sarif.js

```diff
@@ -0,0 +1,194 @@
+const fs = require('fs');
+const url = require('url');
+
+const chardet = require('chardet');
+const utf8 = require('utf8');
+
+function getResultLevel(message) {
+  if (message.fatal || message.severity === 2) {
+    return 'error';
+  }
+  return 'warning';
+}
+
+/**
+ * This printer is heavily based on Microsoft's SARIF formatter for ESLint.
+ * https://github.com/microsoft/sarif-sdk/blob/main/src/ESLint.Formatter/sarif.js
+ */
+class SarifPrinter {
+  constructor(options = {}) {
+    this.options = options;
+    this.console = options.console || console;
+  }
+
+  print(results) {
+    const sarifLog = {
+      version: '2.1.0',
+      $schema: 'http://json.schemastore.org/sarif-2.1.0-rtm.5',
+      runs: [
+        {
+          tool: {
+            driver: {
+              name: 'ember-template-lint',
+              informationUri: 'https://github.com/ember-template-lint/ember-template-lint',
+              rules: [],
+              version: require('../../package.json').version,
+            },
+          },
+        },
+      ],
+    };
+    const sarifFiles = {};
+    const sarifArtifactIndices = {};
+    let nextArtifactIndex = 0;
+    const sarifRules = {};
+    const sarifRuleIndices = {};
+    let nextRuleIndex = 0;
+    const sarifResults = [];
+    const embedFileContents = process.env.SARIF_EMBER_TEMPLATE_LINT_EMBED === 'true';
+
+    const toolConfigurationNotifications = [];
+
+    for (let filePath of Object.keys(results.files)) {
+      let result = results.files[filePath];
+
+      if (typeof sarifFiles[result.filePath] === 'undefined') {
+        sarifArtifactIndices[result.filePath] = nextArtifactIndex++;
+
+        sarifFiles[result.filePath] = {
+          location: {
+            uri: url.pathToFileURL(result.filePath),
+          },
+        };
+
+        let contentsUtf8;
+
+        if (embedFileContents) {
+          try {
+            // Try to get the file contents and encoding.
+            const contents = fs.readFileSync(result.filePath);
+            const encoding = chardet.detect(contents);
+
+            // Encoding will be null if it could not be determined.
+            if (encoding) {
+              // Convert the content bytes to a UTF-8 string.
+              contentsUtf8 = utf8.encode(contents.toString(encoding.encoding));
+
+              sarifFiles[result.filePath].contents = {
+                text: contentsUtf8,
+              };
```

Comment:
I don't fully grok this. We basically only operate on UTF-8 files...

Reply:
Yep, we can absolutely just trim this stuff out. It was largely cribbed from the eslint formatter, and I knew this would be a point of discussion. I wanted to get your 👀 on it to discuss this aspect before I made the assumption it should be `utf-8` always.

Reply:
Removed this.